File: task_equivalents/llm_translations/gpt/base/python/data_visualization/01_scatter_trend_grouped.py
import pandas as pd
import statsmodels.formula.api as smf
import matplotlib.pyplot as plt

auto = pd.read_csv("data/csv/Auto.csv", header=0, na_values="?")
auto = auto.dropna()

model1 = smf.ols("mpg ~ horsepower", data=auto).fit()

plt.scatter(auto["horsepower"], auto["mpg"])
x_vals = auto["horsepower"].sort_values()
y_vals = model1.predict(pd.DataFrame({"horsepower": x_vals}))
plt.plot(x_vals, y_vals, linewidth=3, color="red")
plt.xlabel("horsepower")
plt.ylabel("mpg")
plt.show()

File: task_equivalents/llm_translations/gpt/base/r/data_visualization/01_scatter_trend_grouped.R
library(readr)
library(ggplot2)

Auto <- read_csv("data/csv/Auto.csv", na = "?") %>%
na.omit()

n <- nrow(Auto)
X <- Auto$horsepower
Y <- Auto$mpg

model1_fit <- lm(Y ~ X)

a <- coef(model1_fit)[1]
b <- coef(model1_fit)[2]
yfit <- a + b * X

plot(X, Y, xlab = "horsepower", ylab = "mpg")
lines(X[order(X)], yfit[order(X)], col = "red")